src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,175 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+
+    class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(lastNode, value, null);
+        if (lastNode == null) {
+            firstNode = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        lastNode = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            checkRange(index);
+            Node<T> currentNode = getNode(index);
+            Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+            Node<T> prev = currentNode.prev;
+            newNode.prev = prev;
+            newNode.next = currentNode;
+            currentNode.prev = newNode;
+            if (prev == null) {
+                firstNode = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkRange(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkRange(index);
```

Comment:
move `checkRange(index);` inside of the `getNode()` method to avoid code duplication.
check all code occurrences please